### PR TITLE
case 22189: Add display name field to the log in screen.

### PR DESCRIPTION
**Pull request by SaracenOne**
This simply adds a new field which allows new users logging in to set their display name since new users will not likely be aware that their display name defaults to anonymous and how to change it. By default, it will fill this field with whatever your local display name currently is. Your display name will be changed to whatever is written in this field only when you log in.

https://highfidelity.fogbugz.com/f/cases/22189/Add-display-name-field-to-the-log-in-screen

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Can someone from High Fidelity approve this PR?

**Comment by hifi-gustavo** [comment]
Can someone from High Fidelity approve this PR?

**Comment by SamGondelman** [comment]
add to whitelist

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18676/HighFidelity-Beta-PR15334-67dc2de.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18676/HighFidelity-Beta-Interface-PR15334-67dc2de.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18676/HighFidelity-Beta-PR15334-67dc2de.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18676/HighFidelity-Quest-Beta-PR15334-67dc2de.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18676/HighFidelity-Beta-PR15334-67dc2de.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18676/HighFidelity-Beta-Interface-PR15334-67dc2de.exe

File: interface/resources/qml/LoginDialog/LinkAccountBody.qml [hunk 1]
```diff
@@ -85,27 +85,31 @@ Item {
...
+        displayNameField.placeholderText = "Display Name (optional)";
+        var savedDisplayName = Settings.getValue("Avatar/displayName", "");
+        displayNameField.text = savedDisplayName;
         emailField.placeholderText = "Username or Email";
         var savedUsername = Settings.getValue("keepMeLoggedIn/savedUsername", "");
         emailField.text = keepMeLoggedInCheckbox.checked ? savedUsername === "Unknown user" ? "" : savedUsername : "";
         if (linkAccountBody.linkSteam || linkAccountBody.linkOculus) {
             loginButton.width = (passwordField.width - hifi.dimensions.contentSpacing.x) / 2;
-            loginButton.anchors.right = emailField.right;
+            loginButton.anchors.right = displayName.right;
```
Comment (wayne-chen) [inline interface/resources/qml/LoginDialog/LinkAccountBody.qml]: typo here

File: interface/resources/qml/LoginDialog/LinkAccountBody.qml [hunk 2]
```diff
@@ -85,27 +85,31 @@ Item {
...
         } else {
-            loginButton.anchors.left = emailField.left;
+            loginButton.anchors.left = displayName.left;
```
Comment (wayne-chen) [inline interface/resources/qml/LoginDialog/LinkAccountBody.qml]: typo here

File: interface/resources/qml/LoginDialog/LinkAccountBody.qml (hunk 1 above)
Comment (wayne-chen) [inline interface/resources/qml/LoginDialog/LinkAccountBody.qml]: typo here

File: interface/resources/qml/LoginDialog/LinkAccountBody.qml (hunk 2 above)
Comment (wayne-chen) [inline interface/resources/qml/LoginDialog/LinkAccountBody.qml]: typo here

File: interface/resources/qml/LoginDialog/LinkAccountBody.qml [hunk 3]
```diff
@@ -350,7 +396,7 @@ Item {
                 anchors {
                     top: loginButton.bottom
                     topMargin: hifi.dimensions.contentSpacing.y
-                    left: emailField.left
+                    left: displayName.left
```
Comment (wayne-chen) [inline interface/resources/qml/LoginDialog/LinkAccountBody.qml]: typo here

File: interface/resources/qml/LoginDialog/LinkAccountBody.qml [hunk 4]
```diff
@@ -381,13 +427,13 @@ Item {
             }
             HifiControlsUit.Button {
                 id: continueButton;
-                width: emailField.width;
+                width: displayName.width;
```
Comment (wayne-chen) [inline interface/resources/qml/LoginDialog/LinkAccountBody.qml]: typo here

File: interface/resources/qml/LoginDialog/LinkAccountBody.qml [hunk 5]
```diff
@@ -381,13 +427,13 @@ Item {
...
                 height: d.minHeightButton
                 color: hifi.buttons.none;
                 anchors {
                     top: cantAccessText.bottom
                     topMargin: hifi.dimensions.contentSpacing.y
-                    left: emailField.left
+                    left: displayName.left
```
Comment (wayne-chen) [inline interface/resources/qml/LoginDialog/LinkAccountBody.qml]: typo here

File: interface/resources/qml/LoginDialog/LinkAccountBody.qml [hunk 6]
```diff
@@ -535,14 +582,15 @@ Item {
         onFocusEnabled: {
             if (!linkAccountBody.lostFocus) {
                 Qt.callLater(function() {
-                    emailField.forceActiveFocus();
+                    displayName.forceActiveFocus();
```
Comment (wayne-chen) [inline interface/resources/qml/LoginDialog/LinkAccountBody.qml]: typo here

File: interface/resources/qml/LoginDialog/LinkAccountBody.qml [hunk 7]
```diff
@@ -535,14 +582,15 @@ Item {
...
                 });
             }
         }
         onFocusDisabled: {
             linkAccountBody.lostFocus = !root.isTablet && !root.isOverlay;
             if (linkAccountBody.lostFocus) {
                 Qt.callLater(function() {
+                    displayName.focus = false;
```
Comment (wayne-chen) [inline interface/resources/qml/LoginDialog/LinkAccountBody.qml]: typo here

File: interface/resources/qml/LoginDialog/LinkAccountBody.qml [hunk 8]
```diff
@@ -558,7 +606,7 @@ Item {
         d.resize();
         init();
         Qt.callLater(function() {
-            emailField.forceActiveFocus();
+            displayName.forceActiveFocus();
```
Comment (wayne-chen) [inline interface/resources/qml/LoginDialog/LinkAccountBody.qml]: typo here

**Comment by SaracenOne** [comment]
The change @wayne-chen requested has now been pushed. Apologies for the oversight.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18741/HighFidelity-Beta-PR15334-36af2c8.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18741/HighFidelity-Beta-Interface-PR15334-36af2c8.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18741/HighFidelity-Beta-PR15334-36af2c8.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18741/HighFidelity-Quest-Beta-PR15334-36af2c8.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18741/HighFidelity-Beta-PR15334-36af2c8.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18741/HighFidelity-Beta-Interface-PR15334-36af2c8.exe

**Comment by shanzzam** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21975/HighFidelity-Beta-PR15334-36af2c8.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21975/HighFidelity-Quest-Beta-PR15334-36af2c8.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21975/HighFidelity-Beta-PR15334-36af2c8.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21975/HighFidelity-Beta-Interface-PR15334-36af2c8.dmg

**Comment by shanzzam** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22263/HighFidelity-Beta-PR15334-36af2c8.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22263/HighFidelity-Quest-Beta-PR15334-36af2c8.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22263/HighFidelity-Beta-PR15334-36af2c8.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22263/HighFidelity-Beta-Interface-PR15334-36af2c8.dmg

**Comment by ZappoMan** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22529/HighFidelity-Beta-PR15334-36af2c8.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22529/HighFidelity-Quest-Beta-PR15334-36af2c8.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22529/HighFidelity-Beta-PR15334-36af2c8.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22529/HighFidelity-Beta-Interface-PR15334-36af2c8.dmg

**Comment by shanzzam** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22604/HighFidelity-Beta-PR15334-36af2c8.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22604/HighFidelity-Quest-Beta-PR15334-36af2c8.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22604/HighFidelity-Beta-PR15334-36af2c8.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22604/HighFidelity-Beta-Interface-PR15334-36af2c8.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22604/HighFidelity-Beta-PR15334-36af2c8.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22604/HighFidelity-Beta-Interface-PR15334-36af2c8.exe

**Comment by ZappoMan** [comment]
tested and it behaves as expected/advertised

**Pull request merged**